analysis.tex

```diff
@@ -61,31 +79,202 @@ \section{Analysis}
-\end{theorem}
+Recall that a distribution ensemble is \emph{unpredictable} if no
+polynomial-time adversary can guess its output. The cryptographic
+predictability game is illustrated in Algorithm~\ref{alg.predict-game}.
+
+\import{.}{./algorithms/alg.predict-game.tex}
+
+\begin{definition}[Unpredictable distribution]
+  A distribution ensemble $\{X_{\kappa}\}_{\kappa\in\mathbb{N}}$ is
+  \emph{unpredictable} if for all probabilistic polynomial-time adversaries
+  $\mathcal{A}$ there exists a negligible function $negl$ such that
+  $\Pr[\textsc{predict}_{\mathcal{A},X}(\kappa) = \true] < \negl$.
+\end{definition}
+
+\begin{lemma}[Negligible predictability]\label{lem:negl-pred}
+  Consider a distribution ensemble $\{X_{\kappa}\}_{\kappa\in\mathbb{N}}$ and
+  a negligible function $\negl$. If for all $\kappa$ for all
+  $x \in [X_\kappa]$ we have that
+  $\Pr_{x^* \gets X_\kappa}[x^* = x] \leq \negl$, then
+  $X$ is unpredictable.
+\end{lemma}
 \begin{proof}
-  Let $\mathcal{A}$ be an arbitrary adversary against $\Pi$.
-  We will construct the Collision Resistance adversary $A^*$ against $H$.
+  Consider a probabilistic polynomial-time adversary $\mathcal{A}$ which
+  predicts $X_\kappa$. The adversary is not given any input beyond $1^\kappa$,
+  hence the distribution of its output is independent from the choice of the
+  challenger. Therefore
 
-  TODO
+  \begin{align*}
+  \Pr[\textsc{predict}_{\mathcal{A},X}(\kappa) = \true] &=
+  \sum_{x' \in [X]}\Pr_{x \gets X}[\mathcal{A}(\kappa) = x'\land x = x'] =\\
+  \sum_{x' \in [X]}\Pr[\mathcal{A}(\kappa) = x']\Pr_{x \gets X}[x = x']
+  &\leq \negl\sum_{x' \in [X]}\Pr[\mathcal{A}(\kappa) = x']
+  \leq \negl
+  \end{align*}
+\end{proof}
 
-  From security of signature scheme, calling $\Gen()$ twice will never yield the same $pk$.
+We now posit that no adversary can predict the public key of a secure signature scheme, except with negligible probability.
+
+\begin{lemma}[Public key unpredictability]\label{lem:pk-unpredictability}
+  Let $S = (\textsf{Gen}, \textsf{Sig}, \textsf{Ver})$ be a secure signature scheme.
+  Then the distribution ensemble
+  $X_\kappa = \{(sk, pk) \gets \textsf{Gen}(1^\kappa); pk\}$ is
+  unpredictable.
+\end{lemma}
+\begin{proof}
+  Let $\widehat{pk}$ be an arbitrary $\kappa$-length public key generated
+  by $\textsf{Gen}$ with some probability $p$.
+  Consider the existential forgery adversary $\mathcal{A}$ which works as
+  follows. It receives $pk$ as its input from the challenger, but ignores it
+  and generates a new key pair $(pk', sk') \gets \textsf{Gen}(1^\kappa)$.
+  Since the
+  two invocations of $\textsf{Gen}$ are independent,
+  $\Pr[pk = pk'] \geq \Pr[pk = \widehat{pk} \land pk' = \widehat{pk}]
+  = \Pr[pk = \widehat{pk}]\Pr[pk = \widehat{pk'}] = p^2$. The adversary checks
+  whether $pk = pk'$. If not, it aborts. Otherwise, it uses $sk'$ to sign the
+  message $m = \epsilon$ and returns the forgery $\sigma = \textsf{Sig}(sk, m)$.
```

Comment:
What is \epsilon here?

Reply:
The empty string.

Reply:
I now defined it in the notation section.